src/plugins/data/common/field_formats/field_format.ts

```diff
@@ -160,32 +168,31 @@ export abstract class FieldFormat {
   }
 
   static from(convertFn: FieldFormatConvertFunction) {
-    return createCustomFieldFormat(FieldFormat.toConvertObject(convertFn));
+    return createCustomFieldFormat(convertFn);
   }
 
   private static setupContentType(
     fieldFormat: IFieldFormat,
-    convert: FieldFormatConvert | FieldFormatConvertFunction
+    convert: Partial<FieldFormatConvert> | FieldFormatConvertFunction = {}
   ): FieldFormatConvert {
-    const convertObject = FieldFormat.toConvertObject(convert);
+    const convertObject = isFieldFormatConvertFn(convert)
+      ? FieldFormat.toConvertObject(convert)
+      : convert;
 
     return {
-      ...textContentTypeSetup(fieldFormat, convertObject),
-      ...htmlContentTypeSetup(fieldFormat, convertObject),
+      [TEXT_CONTEXT_TYPE]: textContentTypeSetup(fieldFormat, convertObject),
+      [HTML_CONTEXT_TYPE]: htmlContentTypeSetup(fieldFormat, convertObject),
     };
   }
 
-  private static toConvertObject(
-    convert: FieldFormatConvert | FieldFormatConvertFunction
-  ): FieldFormatConvert {
-    if (isFunction(convert)) {
+  private static toConvertObject(convert: FieldFormatConvertFunction): Partial<FieldFormatConvert> {
+    if (isFieldFormatConvertFn(convert)) {
       return {
-        [DEFAULT_CONTEXT_TYPE]: convert,
+        [TEXT_CONTEXT_TYPE]: convert,
```

Comment:
can we add both 
```
text: convert,
html: convert,
```

this way we can get rid of the logic in consumers where they would need to check if `html` is available and if not call `text`

Reply:
Not sure that we should do it now. We can remove this method in future when all formatters will be refactored to _convert: { text: (value: any): string { ... } } format